cmd/network/network.go

```diff
@@ -11,12 +15,43 @@ import (
 	"github.com/rancher/os/config"
 )
 
+const (
+	NETWORK_DONE     = "/var/run/network.done"
+	WAIT_FOR_NETWORK = "wait-for-network"
+)
+
+func handleTerm() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGTERM)
+	<-c
+	os.Exit(0)
+}
+
+func sendTerm(proc string) {
+	cmd := exec.Command("killall", "-TERM", proc)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	cmd.Run()
+}
+
 func Main() {
 	args := os.Args
 	if len(args) > 1 {
 		fmt.Println("call " + args[0] + " to load network config from cloud-config.yml")
 		return
 	}
+
+	if strings.HasSuffix(args[0], WAIT_FOR_NETWORK) { // we're wait-for-network
+		go handleTerm()
```

Comment:
Can you just make this a separate Main() method?  Seems like there is not real value to embedded it here and then you don't need to check `args[0]` because `registerCmd()` just does that for you.